core/src/main/kotlin/ch/ergon/dope/resolvable/fromable/UseIndex.kt

```diff
@@ -0,0 +1,73 @@
+package ch.ergon.dope.resolvable.fromable
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.formatToQueryStringWithSymbol
+import ch.ergon.dope.validtype.StringType
+
+private const val USE_INDEX = "USE INDEX"
+
+enum class IndexType(val type: String) {
+    USING_GSI("USING GSI"),
+    USING_FTS("USING FTS"),
+}
+
+class IndexReference : Resolvable {
+    private val indexName: TypeExpression<StringType>?
+    private val indexType: IndexType?
+
+    constructor(indexName: TypeExpression<StringType>, indexType: IndexType) {
+        this.indexName = indexName
+        this.indexType = indexType
+    }
+
+    constructor(indexName: TypeExpression<StringType>) {
+        this.indexName = indexName
+        this.indexType = null
+    }
+
+    constructor(indexName: String) {
+        this.indexName = indexName.toDopeType()
+        this.indexType = null
+    }
+
+    constructor(indexType: IndexType) {
```

Comment:
I would assume this will generate something like `... USE INDEX (USING GSI) ...` - is this really correct syntax?

Reply:
Yes it is, for example `SELECT `city` FROM airport USE INDEX (USING GSI)` works perfectly fine